### PR TITLE
Fixes #3533, override Salesforce component class to avoid deadlocks in underlying HTTP client by reverting to use an AsyncCallback

### LINKED ISSUE #3533: Salesforce on-create,on-update not triggered in CR2 prod build
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

I have SF(on-create, lead) -> LOG integration, but the on-create event on field "Lead" is not starting the integration when there is a new lead created in salesforce. This does not work on CR2 prod build, but it is working on 1.4.7 upstream tag.

There is nothing in the logs, just that the integration was started:

```
Starting the Java application using /opt/run-java/run-java.sh ...
exec java -javaagent:/opt/jolokia/jolokia.jar=config=/opt/jolokia/etc/jolokia.properties -javaagent:/opt/prometheus/jmx_prometheus_javaagent.jar=9779:/tmp/src/prometheus-config.yml -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40 -XX:+ExitOnOutOfMemoryError -cp . -jar /deployments/project-0.1-SNAPSHOT.jar
I> No access restrictor found, access to any MBean is allowed
Jolokia: Agent started with URL https://172.17.0.10:8778/jolokia/
      _______.                 _               _
    /       |                 | |             (_)
   |   (----`_   _  ____    _ | |  ____   ___  _   ___
    \   \   | | | ||  _ \  / || | / _  ) /___)| | /___)
.----)   |  | |_| || | | |( (_| |( (/ / |___ || ||___ |
|_______/    \__  ||_| |_| \____| \____)(___/ |_|(___/
============ (____/ ===================================
:: Integration ::  :: v


2018-09-01 08:31:06.718  INFO 1 --- [           main] io.syndesis.example.Application          : Starting Application on i-sf-2-jnmkf with PID 1 (/deployments/project-0.1-SNAPSHOT.jar started by jboss in /deployments)
2018-09-01 08:31:06.732 DEBUG 1 --- [           main] io.syndesis.example.Application          : Running with Spring Boot v1.5.13.RELEASE, Spring v4.3.17.RELEASE
2018-09-01 08:31:06.733  INFO 1 --- [           main] io.syndesis.example.Application          : No active profile set, falling back to default profiles: default
2018-09-01 08:31:06.970  INFO 1 --- [           main] ationConfigEmbeddedWebApplicationContext : Refreshing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@75f9eccc: startup date [Sat Sep 01 08:31:06 UTC 2018]; root of context hierarchy
2018-09-01 08:31:10.297  INFO 1 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.apache.camel.spring.boot.CamelAutoConfiguration' of type [org.apache.camel.spring.boot.CamelAutoConfiguration$$EnhancerBySpringCGLIB$$c70462ee] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-09-01 08:31:11.287  INFO 1 --- [           main] org.xnio                                 : XNIO version 3.3.8.Final
2018-09-01 08:31:11.323  INFO 1 --- [           main] org.xnio.nio                             : XNIO NIO Implementation Version 3.3.8.Final
2018-09-01 08:31:11.398  WARN 1 --- [           main] io.undertow.websockets.jsr               : UT026009: XNIO worker was not set on WebSocketDeploymentInfo, the default worker will be used
2018-09-01 08:31:11.399  WARN 1 --- [           main] io.undertow.websockets.jsr               : UT026010: Buffer pool was not set on WebSocketDeploymentInfo, the default pool will be used
2018-09-01 08:31:11.444  INFO 1 --- [           main] io.undertow.servlet                      : Initializing Spring embedded WebApplicationContext
2018-09-01 08:31:11.445  INFO 1 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 4483 ms
2018-09-01 08:31:11.777  INFO 1 --- [           main] o.s.b.w.servlet.ServletRegistrationBean  : Mapping servlet: 'dispatcherServlet' to [/]
2018-09-01 08:31:11.784  INFO 1 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'metricsFilter' to: [/*]
2018-09-01 08:31:11.786  INFO 1 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'characterEncodingFilter' to: [/*]
2018-09-01 08:31:11.787  INFO 1 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'hiddenHttpMethodFilter' to: [/*]
2018-09-01 08:31:11.787  INFO 1 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'httpPutFormContentFilter' to: [/*]
2018-09-01 08:31:11.787  INFO 1 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'requestContextFilter' to: [/*]
2018-09-01 08:31:11.787  INFO 1 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'webRequestLoggingFilter' to: [/*]
2018-09-01 08:31:11.787  INFO 1 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'applicationContextIdFilter' to: [/*]
2018-09-01 08:31:12.292  INFO 1 --- [           main] s.w.s.m.m.a.RequestMappingHandlerAdapter : Looking for @ControllerAdvice: org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@75f9eccc: startup date [Sat Sep 01 08:31:06 UTC 2018]; root of context hierarchy
2018-09-01 08:31:12.436  INFO 1 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/error],produces=[text/html]}" onto public org.springframework.web.servlet.ModelAndView org.springframework.boot.autoconfigure.web.BasicErrorController.errorHtml(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)
2018-09-01 08:31:12.437  INFO 1 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/error]}" onto public org.springframework.http.ResponseEntity<java.util.Map<java.lang.String, java.lang.Object>> org.springframework.boot.autoconfigure.web.BasicErrorController.error(javax.servlet.http.HttpServletRequest)
2018-09-01 08:31:12.485  INFO 1 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped URL path [/webjars/**] onto handler of type [class org.springframework.web.servlet.resource.ResourceHttpRequestHandler]
2018-09-01 08:31:12.485  INFO 1 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped URL path [/**] onto handler of type [class org.springframework.web.servlet.resource.ResourceHttpRequestHandler]
2018-09-01 08:31:12.554  INFO 1 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped URL path [/**/favicon.ico] onto handler of type [class org.springframework.web.servlet.resource.ResourceHttpRequestHandler]
2018-09-01 08:31:13.048  INFO 1 --- [           main] o.a.c.s.boot.CamelAutoConfiguration      : Using custom InterceptStrategy with id: integrationLoggingInterceptStrategy and implementation: io.syndesis.integration.runtime.logging.ActivityTrackingInterceptStrategy@6eeade6c
2018-09-01 08:31:13.342  INFO 1 --- [           main] o.a.c.i.converter.DefaultTypeConverter   : Type converters loaded (core: 194, classpath: 2)
2018-09-01 08:31:14.596  INFO 1 --- [           main] o.s.b.a.e.mvc.EndpointHandlerMapping     : Mapped "{[/health || /health.json],methods=[GET],produces=[application/vnd.spring-boot.actuator.v1+json || application/json]}" onto public java.lang.Object org.springframework.boot.actuate.endpoint.mvc.HealthMvcEndpoint.invoke(javax.servlet.http.HttpServletRequest,java.security.Principal)
2018-09-01 08:31:14.864  INFO 1 --- [           main] o.s.j.e.a.AnnotationMBeanExporter        : Registering beans for JMX exposure on startup
2018-09-01 08:31:14.881  INFO 1 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Starting beans in phase 0
2018-09-01 08:31:14.925 DEBUG 1 --- [           main] o.a.camel.spring.boot.RoutesCollector    : Post-processing CamelContext bean: sf
2018-09-01 08:31:14.936 DEBUG 1 --- [           main] o.a.camel.spring.boot.RoutesCollector    : CamelContextConfiguration found. Invoking beforeApplicationStart: io.syndesis.integration.runtime.IntegrationRuntimeAutoConfiguration$1@4c1f22f3
2018-09-01 08:31:14.998  INFO 1 --- [           main] i.s.i.runtime.IntegrationRouteBuilder    : Loading integration from: classpath:syndesis/integration/integration.json
2018-09-01 08:31:16.344 DEBUG 1 --- [           main] i.s.i.runtime.IntegrationRouteBuilder    : Step kind: endpoint, handler: io.syndesis.integration.runtime.handlers.ConnectorStepHandler
2018-09-01 08:31:16.484  INFO 1 --- [           main] i.s.i.c.proxy.ComponentProxyComponent    : Register component: salesforce-1 (type: org.apache.camel.component.salesforce.SalesforceComponent) with scheme: salesforce and alias: salesforce-salesforce-1
2018-09-01 08:31:16.497  INFO 1 --- [           main] org.eclipse.jetty.util.log               : Logging initialized @12677ms to org.eclipse.jetty.util.log.Slf4jLog
2018-09-01 08:31:16.500  INFO 1 --- [           main] o.a.c.util.jsse.SSLContextParameters     : Available providers: SUN version 1.8.
2018-09-01 08:31:16.712  INFO 1 --- [           main] o.a.c.c.s.internal.SalesforceSession     : Login at Salesforce loginUrl: https://login.salesforce.com/services/oauth2/token
2018-09-01 08:31:17.643  INFO 1 --- [           main] o.a.c.c.s.internal.SalesforceSession     : Login successful
2018-09-01 08:31:17.648  WARN 1 --- [           main] o.a.c.c.salesforce.SalesforceComponent   : Missing property packages, getSObject* operations will NOT work without property rawPayload=true
2018-09-01 08:31:17.650 DEBUG 1 --- [           main] i.s.i.c.proxy.ComponentProxyComponent    : Starting connector: salesforce-1
2018-09-01 08:31:17.728 DEBUG 1 --- [           main] i.s.i.runtime.IntegrationRouteBuilder    : Step kind: endpoint, handler: io.syndesis.integration.runtime.handlers.ConnectorStepHandler
2018-09-01 08:31:17.749 DEBUG 1 --- [           main] i.s.i.c.proxy.ComponentProxyComponent    : Starting connector: log-2
2018-09-01 08:31:17.758 DEBUG 1 --- [           main] o.a.camel.spring.boot.RoutesCollector    : CamelContextConfiguration found. Invoking beforeApplicationStart: io.syndesis.integration.runtime.jmx.IntegrationMetadataAutoConfiguration$1@749c877b
2018-09-01 08:31:17.760  INFO 1 --- [           main] r.j.IntegrationMetadataAutoConfiguration : Added Syndesis MBean Service
2018-09-01 08:31:17.760 DEBUG 1 --- [           main] o.a.camel.spring.boot.RoutesCollector    : CamelContextConfiguration found. Invoking beforeApplicationStart: io.syndesis.integration.runtime.logging.IntegrationLoggingAutoConfiguration$1@2b95e48b
2018-09-01 08:31:17.805  INFO 1 --- [           main] o.a.camel.spring.SpringCamelContext      : Apache Camel 2.21.0.fuse-710018-redhat-00001 (CamelContext: sf) is starting
2018-09-01 08:31:17.811  INFO 1 --- [           main] o.a.camel.spring.SpringCamelContext      : StreamCaching is enabled on CamelContext: sf
2018-09-01 08:31:17.828  INFO 1 --- [           main] o.a.c.m.ManagedManagementStrategy        : JMX is enabled
2018-09-01 08:31:18.195  INFO 1 --- [           main] i.s.i.c.proxy.ComponentProxyComponent    : Connector resolved: salesforce-1 -> salesforce-salesforce-1://syndesis_Lead_c?notifyForFields=ALL&notifyForOperationCreate=true&notifyForOperationDelete=false&notifyForOperationUndelete=false&notifyForOperationUpdate=false&sObjectClass=io.syndesis.connector.salesforce.SalesforceIdentifier&sObjectQuery=SELECT+Id+FROM+Lead&updateTopic=true
2018-09-01 08:31:18.349  INFO 1 --- [           main] i.s.i.c.proxy.ComponentProxyComponent    : Connector resolved: log-2 -> log://syndesis-log?level=INFO&showAll=true
2018-09-01 08:31:18.385  INFO 1 --- [           main] o.a.c.impl.DefaultStreamCachingStrategy  : StreamCaching in use with spool directory: /tmp/camel/camel-tmp-b30e4464-019d-4b26-9b46-c3c35bccb8c3 and rules: [Spool > 128K body size]
2018-09-01 08:31:24.044  INFO 1 --- [           main] i.s.i.r.jmx.CamelContextMetadataMBean    : Registered mbean io.syndesis.camel:context=sf,type=context,name="sf"
2018-09-01 08:31:24.583  INFO 1 --- [           main] o.a.c.c.s.i.streaming.PushTopicHelper    : Found existing topic syndesis_Lead_c: {"attributes":{"type":"PushTopic","url":"/services/data/v34.0/sobjects/PushTopic/0IF0Y000000Xl3dWAC"},"Query":"SELECT Id FROM Lead","Description":"Topic created by Camel Salesforce component","NotifyForOperationDelete":false,"NotifyForOperationUndelete":false,"IsActive":true,"NotifyForFields":"All","NotifyForOperations":"Create","NotifyForOperationCreate":true,"NotifyForOperationUpdate":false,"ApiVersion":34.0,"Id":"0IF0Y000000Xl3dWAC","Name":"syndesis_Lead_c"}
2018-09-01 08:31:24.594  INFO 1 --- [           main] o.a.c.c.s.i.s.SubscriptionHelper         : Subscribing to channel /topic/syndesis_Lead_c...
2018-09-01 08:31:24.601  INFO 1 --- [           main] o.a.camel.spring.SpringCamelContext      : Route: -LLJF5Vx_TDQe8bCwwcq started and consuming from: salesforce-salesforce-1://syndesis_Lead_c?notifyForFields=ALL&notifyForOperationCreate=true&notifyForOperationDelete=false&notifyForOperationUndelete=false&notifyForOperationUpdate=false&sObjectClass=io.syndesis.connector.salesforce.SalesforceIdentifier&sObjectQuery=SELECT+Id+FROM+Lead&updateTopic=true
2018-09-01 08:31:24.604  INFO 1 --- [           main] o.a.camel.spring.SpringCamelContext      : Total 1 routes, of which 1 are started
2018-09-01 08:31:24.634  INFO 1 --- [           main] o.a.camel.spring.SpringCamelContext      : Apache Camel 2.21.0.fuse-710018-redhat-00001 (CamelContext: sf) started in 6.836 seconds
2018-09-01 08:31:24.637 DEBUG 1 --- [           main] o.a.camel.spring.boot.RoutesCollector    : CamelContextConfiguration found. Invoking afterApplicationStart: io.syndesis.integration.runtime.IntegrationRuntimeAutoConfiguration$1@4c1f22f3
2018-09-01 08:31:24.707  INFO 1 --- [ent@25ad25f5-29] o.a.c.c.s.i.s.SubscriptionHelper         : Subscribed to channel /topic/syndesis_Lead_c
2018-09-01 08:31:25.764 DEBUG 1 --- [           main] .i.r.IntegrationRuntimeAutoConfiguration : Routes: 
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<routes xmlns="http://camel.apache.org/schema/spring">
    <route customId="true" id="-LLJF5Vx_TDQe8bCwwcq">
        <from customId="true" id="-LLJF5Vx_TDQe8bCwwcq" uri="salesforce-1"/>
        <setHeader headerName="Syndesis.STEP_ID" id="setHeader1">
            <constant>-LLJF5Vx_TDQe8bCwwcq</constant>
        </setHeader>
        <process id="process1"/>
        <pipeline customId="true" id="step:-LLJF8Wp_TDQe8bCwwcq">
            <setHeader headerName="Syndesis.STEP_ID" id="setHeader2">
                <constant>-LLJF8Wp_TDQe8bCwwcq</constant>
            </setHeader>
            <to id="to1" uri="log-2"/>
            <process id="process2"/>
        </pipeline>
    </route>
</routes>

2018-09-01 08:31:25.764 DEBUG 1 --- [           main] o.a.camel.spring.boot.RoutesCollector    : CamelContextConfiguration found. Invoking afterApplicationStart: io.syndesis.integration.runtime.jmx.IntegrationMetadataAutoConfiguration$1@749c877b
2018-09-01 08:31:25.764 DEBUG 1 --- [           main] o.a.camel.spring.boot.RoutesCollector    : CamelContextConfiguration found. Invoking afterApplicationStart: io.syndesis.integration.runtime.logging.IntegrationLoggingAutoConfiguration$1@2b95e48b
2018-09-01 08:31:25.864  INFO 1 --- [           main] b.c.e.u.UndertowEmbeddedServletContainer : Undertow started on port(s) 8080 (http)
2018-09-01 08:31:25.880  INFO 1 --- [           main] io.syndesis.example.Application          : Started Application in 20.446 seconds (JVM running for 22.061)
```

## Tasks involved / Steps to Reproduce

1. deploy cr2
2. create sf connection
3. deploy sf (on-create, lead) -> log integration
4. create lead in sf
5. the integration is not triggered

### PR DESCRIPTION
**Pull request by dhirajsb**

### CODE REVIEW

File: app/pom.xml [hunk 1]
```diff
@@ -75,7 +75,7 @@
     <errorprone.version>2.3.1</errorprone.version>
 
     <!-- Global camel version used everywhere -->
-    <camel.version>2.21.0.fuse-000094</camel.version>
+    <camel.version>2.21.0.fuse-710018</camel.version>
```
Comment (apodhrad) [inline app/pom.xml]: Hi @dhirajsb, is this correct camel version? The productized version is **2.21.0.fuse-710018-redhat-00001** as stated in this jira [comment](https://issues.jboss.org/browse/ENTESB-9336?focusedCommentId=13626027&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13626027)

Comment (oscerd) [reply app/pom.xml]: That's the pinned version on Origin nexus, we can't use the redhat version upstream.

Comment (apodhrad) [reply app/pom.xml]: Just wondering, will it be changed during the build process by PNC?

Comment (oscerd) [reply app/pom.xml]: Yes it will

Comment (apodhrad) [reply app/pom.xml]: thank you

Comment (dhirajsb) [reply app/pom.xml]: @apodhrad that's a prod build pinned in maven central, but it's pretty close to the one you mentioned, which is only in RH repos.

Comment (dhirajsb) [reply app/pom.xml]: In any case, that version is overwritten/fixed in the prod build process.

**Pull request merged**